notifications.yaml

```diff
@@ -0,0 +1,220 @@
+swagger: "2.0"
+info: 
+  version: "0.0.1"
+  title: "Notification API"
+  description: "Notification API."
+host: "localhost"
+basePath: "/api/v1"
+schemes: 
+  - "https"
+paths: 
+  /Messages/{id}: 
+    parameters: 
+      - 
+        name: "id"
+        in: "path"
+        required: true
+        type: "string"
+        format: "uuid"
+    get: 
+      operationId: "getMessage"
+      summary: "Get Message"
+      tags: 
+        - "Messages"
+      description: "Retrieve a message already submitted for delivery."
+      responses: 
+        200: 
+          description: "Message found."
+          schema: 
+            $ref: "#/definitions/Message"
+  /Users/{id}/Messages: 
+    parameters: 
+      - 
+        name: "id"
+        in: "path"
+        required: true
+        type: "string"
+        format: "uuid"
+    get: 
+      operationId: "listMessagesByUser"
+      summary: "List Messages by user"
+      tags: 
+        - "Messages"
+      description: "Retrive all messages for this user. \n\nMessages will be returned in inverse delivery order (from last to first)"
+      responses: 
+        200: 
+          description: "Found."
+          schema: 
+            $ref: "#/definitions/MessageCollection"
+  /Messages: 
+    post: 
+      operationId: "sendMessage"
+      summary: "Send Message"
+      tags: 
+        - "Messages"
+      description: "Submit a message for delivery."
+      parameters: 
+        - 
+          name: "message"
+          in: "body"
+          schema: 
+            $ref: "#/definitions/MessageDefinition"
+      responses: 
+        202: 
```

Comment:
202 assumes the request is asynchronous and will be completed later (i.e. the message will be created later)

in our case the creation is synchronous since we're returning the UUID